tfe/resource_tfe_workspace.go

```diff
@@ -671,7 +676,29 @@ func resourceTFEWorkspaceDelete(d *schema.ResourceData, meta interface{}) error
 	id := d.Id()
 
 	log.Printf("[DEBUG] Delete workspace %s", id)
-	err := tfeClient.Workspaces.DeleteByID(ctx, id)
+
+	ws, err := tfeClient.Workspaces.ReadByID(ctx, id)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			return nil
+		}
+		return fmt.Errorf(
+			"Error reading workspace %s: %w", id, err)
+	}
+
+	forceDelete := d.Get("force_delete").(bool)
+
+	if ws.Permissions.CanForceDelete == nil && !forceDelete {
+		return fmt.Errorf(
+			"Error deleting workspace %s: This workspace must be force deleted by setting force_delete=true", id)
```

Comment:
I would find this error to be very helpful in the case when there are managed resources. The API error only mentions doing a destroy run, but that's slightly confusing in the context of the provider config.

Here's what I see:
```
╷
│ Error: Error deleting workspace ws-8ZdBVKREwypBnTpi: conflict
│
│ Workspace is currently managing 1 resources. A destroy run must be applied to remove managed resources before the workspace can be safely deleted
│
│
```

And maybe what I'd like to see if CanForceDelete is true:

```
╷
│ Error: Error deleting workspace ws-8ZdBVKREwypBnTpi: conflict
│
│ Workspace is currently managing 1 resources. A destroy run must be applied to remove managed resources before the workspace can be safely deleted
| 
│ To delete this workspace without destroying the underlying resources, add force_delete = true to the tfe_resource.foo config
│
```

Reply:
I like the additional bit of error text @brandonc suggested

Another thought: this error right here only gets shown when talking to an old TFE and the user tries to delete a workspace without `force_delete=true`

If the user is talking to an up-to-date TFE/TFC and they try to delete a RUMful workspace without `force_delete=true`, they would just see the error returned by the API (the "Workspace is currently managing 1 resource..." bit)...I wonder if, in [this error handling block ](https://github.com/hashicorp/terraform-provider-tfe/pull/675/files#diff-e727ae216721eb3c1f24911e8b1ececae3ae9208f19b886eb627c76236471e13R702), we should detect if:
- it was a safe delete which was attempted and,
- the error is a `conflict`, and
- CanForceDelete is true

in which case we could also append Brandon's suggested error text (the "To delete this workspace without destroying the managed resources..." bit) to the outputted error